e2e/test/iothub/command/CommandE2ETests.cs

```diff
@@ -0,0 +1,438 @@
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Mqtt_Tcp_Only, SetModuleReceiveCommandDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        //[LoggedTestMethod]
+        public async Task Method_ModuleReceivesCommandAndResponseWithDefaultMethodHandler_MqttWs()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Mqtt_WebSocket_Only, SetModuleReceiveCommandDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        //[LoggedTestMethod]
+        public async Task Method_ModuleReceivesCommandAndResponse_Amqp()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Amqp_Tcp_Only, SetModuleReceiveCommandAsync).ConfigureAwait(false);
+        }
+
+        //[LoggedTestMethod]
+        public async Task Method_ModuleReceivesCommandAndResponse_AmqpWs()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Amqp_WebSocket_Only, SetModuleReceiveCommandAsync).ConfigureAwait(false);
+        }
+
+        //[LoggedTestMethod]
+        public async Task Method_ModuleReceivesCommandAndResponseWithDefaultMethodHandler_Amqp()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Amqp_Tcp_Only, SetModuleReceiveCommandDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        //[LoggedTestMethod]
+        public async Task Method_ModuleReceivesCommandAndResponseWithDefaultMethodHandler_AmqpWs()
+        {
+            await SendCommandAndRespondAsync(Client.TransportType.Amqp_WebSocket_Only, SetModuleReceiveCommandDefaultHandlerAsync).ConfigureAwait(false);
+        }
+
+        //[LoggedTestMethod]
+        public async Task Method_ServiceInvokeDeviceCommandWithUnknownModuleThrows()
+        {
+            // setup
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, "ModuleNotFoundTest").ConfigureAwait(false);
+            using var serviceClient = ServiceClient.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            var methodInvocation = new CloudToDeviceMethod("SetTelemetryInterval");
+            methodInvocation.SetPayloadJson("10");
+
+            // act
+            ErrorCode actualErrorCode = ErrorCode.InvalidErrorCode;
+            try
+            {
+                // Invoke the direct method asynchronously and get the response from the simulated device.
+                await serviceClient.InvokeDeviceMethodAsync(testDevice.Id, "someNonExistantModuleOnAnExistingDevice", methodInvocation).ConfigureAwait(false);
+            }
+            catch (DeviceNotFoundException ex)
+            {
+                // Although the exception is called "Device" not found, it is used for all 404's, including the 404010 that denotes a module was not found
+                actualErrorCode = ex.Code;
+            }
+
+            Assert.AreEqual(ErrorCode.ModuleNotFound, actualErrorCode);
+
+            await serviceClient.CloseAsync().ConfigureAwait(false);
+        }
```

Comment:
Ignore this chunk, leaving these in for when we do the module and service clients.